modules/standard/IO.chpl

```diff
@@ -1399,44 +1404,93 @@ extern type fdflag_t = c_int;
 //  QIO_HINT_BANDWIDTH,
 //  QIO_HINT_CACHED,
 //  QIO_HINT_NOREUSE
-/*
 
-A value of the :type:`iohints` type defines a set of hints about the I/O that
-the file or channel will perform.  These hints may be used by the
-implementation to select optimized versions of the I/O operations.
+deprecated "the 'iohints' type is deprecated; please use the 'ioHintSet' type instead."
+extern type iohints = c_int;
+
+private const IOHINTS_NONE:        c_int = 0;
+private const IOHINTS_SEQUENTIAL:  c_int = QIO_HINT_SEQUENTIAL;
+private const IOHINTS_RANDOM:      c_int = QIO_HINT_RANDOM;
+private const IOHINTS_PREFETCH:    c_int = QIO_HINT_CACHED;
+private const IOHINTS_MMAP:        c_int = QIO_METHOD_MMAP;
+
+/* A value of the :record:`ioHintSet` type defines a set of hints about /
+  the I/O that the file or channel will perform.  These hints may be used
+  by the implementation to select optimized versions of the I/O operations.
+
+  Most hints have POSIX equivalents associated with posix_fadvise() and
+  posix_madvise().
 
-The :type:`iohints` type is implementation-defined.
-The following :type:`iohints` constants are provided:
+  .. code-block:: chapel
 
-  * :const:`IOHINT_NONE` defines an empty set, which provides no hints.
-  * :const:`IOHINT_RANDOM` suggests to expect random access.
-  * :const:`IOHINT_SEQUENTIAL` suggests to expect sequential access.
-  * :const:`IOHINT_CACHED` suggests that the file data is or should be
-    cached in memory, possibly all at once.
-  * :const:`IOHINT_PARALLEL` suggests to expect many channels
-    working with this file in parallel.
+    use IO;
+
+    // define a set of hints using a union
```

Comment:
I'd say "union operation" or "the `or` operator" since `union` is actually a type in Chapel (albeit an unstable one that needs more attention before anyone should really use it)